hibernate-core/src/main/java/org/hibernate/dialect/SybaseASE15Dialect.java

```diff
@@ -30,6 +30,8 @@
 	public SybaseASE15Dialect() {
 		super();
 
+		registerColumnType( Types.VARBINARY, "image" );
+		registerColumnType( Types.VARBINARY, 8000, "varbinary($l)" );
```

Comment:
`8000` is the max length for SQLServer. Should we use `32767` instead here?

Reply:
That's what I decided it should be for Sybase Anywhere. (But apparently for ASE it depends on the version.)

Reply:
Actually it also depends on the page size which can be as small as 2k.

Reply:
Yep, it's based on the database page size which is configurable.

On reflection, we should look up and use the actual page size instead of a fixed value of `8000`.

I couldn't find a quick way to do this after looking through the other dialects.

Can you give any pointers to how this would be done? I would be happy to give that a go.